src/applications/disability-benefits/686/config/form.js

```diff
@@ -88,6 +189,168 @@ function createSpouseLabelSelector(nameTemplate) {
+//
+// ex: given 'marriages[INDEX].locationOfMarriage' and `0` return
+// 'marriages[0].locationOfMarriage'
+function insertRealIndexInKey(key, index) {
+  return key.replace('[INDEX]', `[${index}]`);
+}
+
+// pass in the key so the address we care about can be pulled out of the
+// entire formData object
+function createAddressUISchemaForKey(key, isRequiredCallback = () => true) {
+  return {
+    country: {
+      'ui:title': 'Country',
+      'ui:required': isRequiredCallback,
+    },
+    countryText: {
+      'ui:title': 'Enter Country',
+      'ui:required': (formData, index) =>
+        isInternationalAddressText(
+          formData,
+          `${insertRealIndexInKey(key, index)}.country`,
+        ),
+      'ui:options': {
+        hideIf: (formData, index) =>
+          isNotInternationalAddressText(
+            formData,
+            `${insertRealIndexInKey(key, index)}.country`,
+          ),
+      },
+    },
+    street: {
+      'ui:title': 'Street',
+      'ui:required': isRequiredCallback,
+      'ui:errorMessages': {
+        required: 'Please fill in a valid street address',
+      },
+    },
+    street2: {
+      'ui:title': 'Line 2',
+    },
+    street3: {
+      'ui:title': 'Line 3',
+    },
+    city: {
+      'ui:title': 'City',
+      'ui:required': isRequiredCallback,
+    },
+    state: {
+      'ui:title': 'State',
+      'ui:required': (formData, index) =>
+        isDomesticAddress(
+          formData,
+          `${insertRealIndexInKey(key, index)}.country`,
+        ),
+      'ui:options': {
+        hideIf: (formData, index) =>
+          !isUSAAddress(
+            formData,
+            `${insertRealIndexInKey(key, index)}.country`,
+          ),
+      },
+    },
+    postOffice: {
+      'ui:title': 'Post Office',
+      'ui:required': (formData, index) =>
+        isMilitaryAddress(
+          formData,
+          `${insertRealIndexInKey(key, index)}.state`,
+        ),
+      'ui:options': {
+        hideIf: (formData, index) =>
+          isNotMilitaryAddress(
+            formData,
+            `${insertRealIndexInKey(key, index)}.state`,
+          ),
+      },
+    },
+    postalType: {
+      'ui:title': 'Postal Type',
+      'ui:required': (formData, index) =>
+        isMilitaryAddress(
+          formData,
+          `${insertRealIndexInKey(key, index)}.state`,
+        ),
+      'ui:options': {
+        hideIf: (formData, index) =>
+          isNotMilitaryAddress(
+            formData,
+            `${insertRealIndexInKey(key, index)}.state`,
+          ),
+      },
+    },
+    postalCode: {
+      'ui:title': 'Postal Code',
+      'ui:required': (formData, index) =>
+        isUSAAddress(formData, `${insertRealIndexInKey(key, index)}.country`),
+      'ui:errorMessages': {
+        pattern: 'Please fill in a valid postal code',
+        required: 'Please fill in a valid postal code',
+      },
+    },
+  };
+}
+
+function createLocationUISchemaForKey(key, isRequiredCallback = () => true) {
```

Comment:
We may want to move some of these helpers into the helpers file.

Reply:
Done